debezium-connector-ibmi/src/main/java/io/debezium/connector/db2as400/As400ConnectorTask.java

```diff
@@ -25,175 +25,164 @@
-
-    private Configuration addDefaultHeartbeatToConfig(Configuration config) {
-        int heartbeat = config.getInteger("heartbeat.interval.ms", 0);
-        if (heartbeat == 0) {
-            config = Configuration.copy(config).with("heartbeat.interval.ms", 60000).build();
-        }
-        return config;
-    }
-    
-    private Configuration addDefaultPrefixToConfig(Configuration config) {
-        String prefix = config.getString("topic.prefix", "");
-        if (prefix.isEmpty()) {
-            config = Configuration.copy(config).with("topic.prefix", config.getString(JdbcConfiguration.HOSTNAME)).build();
-        }
-        return config;
-    }
-
-    private Set<String> additionalTablesInConfigTables(final As400ConnectorConfig connectorConfig,
-                                                       As400OffsetContext previousOffset, As400ConnectorConfig newConfig) {
-        String newInclude = newConfig.tableIncludeList();
-        String oldInclude = previousOffset.getIncludeTables();
-        log.info("previous includes {} , new includes {}", oldInclude, newInclude);
-
-        if (oldInclude != null && newInclude != null) {
-            Set<String> newset = Stream.of(newInclude.split(",", -1)).collect(Collectors.toCollection(HashSet::new));
-            newset.removeAll(Set.of(oldInclude.split(",")));
-            return newset;
-        }
-        return Collections.emptySet();
-    }
-
-    @Override
-    protected List<SourceRecord> doPoll() throws InterruptedException {
-        final List<DataChangeEvent> records = queue.poll();
-
-        final List<SourceRecord> sourceRecords = records.stream()
-                .map(DataChangeEvent::getRecord)
-                .collect(Collectors.toList());
-
-        return sourceRecords;
-    }
-
-    @Override
-    protected void doStop() {
-    }
-
-    @Override
-    protected Iterable<Field> getAllConfigurationFields() {
-        return As400ConnectorConfig.ALL_FIELDS;
-    }
+	private static final Logger log = LoggerFactory.getLogger(As400ConnectorTask.class);
+	private volatile ChangeEventQueue<DataChangeEvent> queue;
+	private static final String CONTEXT_NAME = "db2as400-server-connector-task";
+	private As400DatabaseSchema schema;
+
+	@Override
+	public String version() {
+		return Module.version();
+	}
+
+	@Override
+	protected ChangeEventSourceCoordinator<As400Partition, As400OffsetContext> start(Configuration config) {
+		log.warn("starting connector task {}", version());
+		config = addDefaultHeartbeatToConfig(config);
+		config = addDefaultPrefixToConfig(config);
+		final As400ConnectorConfig connectorConfig = new As400ConnectorConfig(config);
+		@SuppressWarnings("unchecked")
+		final TopicNamingStrategy<TableId> topicNamingStrategy = connectorConfig
+				.getTopicNamingStrategy(As400ConnectorConfig.TOPIC_NAMING_STRATEGY, true);
+
+		final SchemaNameAdjuster schemaNameAdjuster = SchemaNameAdjuster.AVRO_FIELD_NAMER;
+
+		// TODO get list of DB ids - see Db2TaskContext
```

Comment:
I know this isn't a new comment... but is this something that still needs to be done?

Reply:
this was old but good call.